Modules/System/AL Rest Client/src/ALRestClientImpl.Codeunit.al

```diff
@@ -0,0 +1,153 @@
+codeunit 2351 "AL Rest Client Impl."
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        HttpAuthentication: Interface "Http Authentication";
+        HttpClient: HttpClient;
+        ConnectionError: Boolean;
+        Initialized: Boolean;
+        NotInitializedErr: Label 'The AL Rest Client has not been initialized';
+        EnvironmentBlocksErr: Label 'Environment blocks an outgoing HTTP request to ''%1''.', Comment = '%1 = url, e.g. https://microsoft.com';
+        ConnectionErr: Label 'Connection to the remote service ''%1'' could not be established.', Comment = '%1 = url, e.g. https://microsoft.com';
+        RequestFailedErr: Label 'The request failed: %1 %2', Comment = '%1 = HTTP status code, %2 = Reason phrase';
+        UserAgentLbl: Label 'Dynamics 365 Business Central - |%1| %2/%3', Locked = true, Comment = '%1 = App Publisher; %2 = App Name; %3 = App Version';
+
+    procedure Initialize()
+    begin
+        ClearAll();
+
+        HttpClient.Clear();
+        HttpClient.Timeout := 60000;
```

Comment:
Here the Timeout is hardcoded (without explanation for this value) and `SetTimeout` is internal, so there's no way to overwrite it.

Reply:
I agree. The default is 100.000 (100 seconds), there is actually no reason to set it to another value.
I've added functions to set and get the timeout.